### STATISTICAL Tests - Assignment

#### Importing necessary Libraries

In [2]:
import numpy as np
import pandas as pd
import scipy.stats as stats
import matplotlib.pyplot as plt
%matplotlib inline
import math

### 1.A physician is evaluating a new diet for her patients with a family history of heart disease. To test the effectiveness of this diet, 16 patients are placed on the diet for 6 months. Their weights and triglyceride levels are measured before and after the study, and the physician wants to know if either set of measurements has changed. (Data set: dietstudy.csv)

In [3]:
#Importing Datasets "dietstudy.csv" and assigning it to the variable dietstudy
dietstudy = pd.read_csv("E:Datasets/dietstudy.csv")
dietstudy

patid  age  gender  tg0  tg1  tg2  tg3  tg4  wgt0  wgt1  wgt2  wgt3  wgt4
0       1   45    Male  180  148  106  113  100   198   196   193   188   192
1       2   56    Male  139   94  119   75   92   237   233   232   228   225
2       3   50    Male  152  185   86  149  118   233   231   229   228   226
3       4   46  Female  112  145  136  149   82   179   181   177   174   172
4       5   64    Male  156  104  157   79   97   219   217   215   213   214
5       6   49  Female  167  138   88  107  171   169   166   165   162   161
6       7   63    Male  138  132  146  143  132   222   219   215   215   210
7       8   63  Female  160  128  150  118  123   167   167   166   162   161
8       9   52    Male  107  120  129  195  174   199   200   196   196   193
9      10   45    Male  156  103  126  135   92   233   229   229   229   226
10     11   61  Female   94  144  114  114  121   179   181   176   173   173
11     12   49  Female  107   93  156  148  150   158   153   155   155   154
12     13   61  Female  145  107  129   86  159   157   151   150   145   143
13     14   59    Male  186  142  128  122  101   216   213   210   210   206
14     15   52    Male  112  107  103   89  148   257   255   254   252   249
15     16   60  Female  104  103  117   79  130   151   146   144   144   140

 ### Two Sample T-Test(Paired)

In [8]:
tgmean1 = dietstudy.tg0.mean()

In [10]:
tgmean2 = dietstudy.tg4.mean()

In [28]:
diff1 = tgmean1 - tgmean2
diff1

14.0625

In [29]:
wgtmean1 = dietstudy.wgt0.mean()

In [30]:
wgtmean2 = dietstudy.wgt4.mean()

In [33]:
diff2 = wgtmean1 - wgtmean2
diff2

8.0625

In [34]:
print("The Initial Triglyceride levels of patients were {}".format(tgmean1))

The Initial Triglyceride levels of patients were 138.4375


In [35]:
print("The Final Triglyceride levels of patients were {}".format(tgmean2))

The Final Triglyceride levels of patients were 124.375


In [36]:
print("The Weights of patients were {}".format(wgtmean1))

The Weights of patients were 198.375


In [37]:
print("The Final Weights of patients were {}".format(wgtmean2))

The Final Weights of patients were 190.3125


### Stage - 1

### H0 : Levels of Triglycerides of individual before diet == Levels of Triglycerides of individual after diet
### H1 : Levels of Triglycerides of individual before diet != Levels of Triglycerides of individual after diet

In [39]:
#Stage 1
stage1_trig = stats.ttest_rel(a=dietstudy.tg0,b=dietstudy.tg4)

#a = triglyceride levels before diet
#b = triglyceride levels after diet

In [41]:
stage1_trig.pvalue

0.24874946576903698

In [42]:
stage1_trig.pvalue > 0.05

True

- Since stage1_trig.pvalue is greater than 0.05 we accept Null hypothesis(H0)

### Stage - 2

### H0 : Weights of individual before diet == Weights of individual after diet
### H1 :Weights of individual before diet != Weights of individual after diet

In [44]:
stage2_weights = stats.ttest_rel(a=dietstudy.wgt0,b=dietstudy.wgt4)

#a = weights before diet
#b = weights after diet

In [46]:
stage2_weights.pvalue

1.137689414996614e-08

In [48]:
stage2_weights.pvalue > 0.05

False

- Since stage2_weights.pvalue is smaller than 0.05 we reject Null hypothesis(H0)

### Conclusions

### 2. An analyst at a department store wants to evaluate a recent credit card promotion. To this end, 500 cardholders were randomly selected. Half received an ad promoting a reduced interest rate on purchases made over the next three months, and half received a standard seasonal ad. Is the promotion effective to increase sales? (Data set: creditpromo.csv)


In [51]:
#Importing Datasets "creditpromo.csv" and assigning it to the variable cred_prom
cred_prom = pd.read_csv("E:Datasets/creditpromo.csv")
cred_prom

id         insert      dollars
0       148       Standard  2232.771979
1       572  New Promotion  1403.807542
2       973       Standard  2327.092181
3      1096       Standard  1280.030541
4      1541  New Promotion  1513.563200
..      ...            ...          ...
495  130163  New Promotion  1513.060134
496  130204  New Promotion  1020.757943
497  130255       Standard  1919.856304
498  130583       Standard  1863.014814
499  130682       Standard  1295.012183

[500 rows x 3 columns]

In [54]:
standard = cred_prom["dollars"].loc[cred_prom["insert"]=="Standard"]
standard

0      2232.771979
2      2327.092181
3      1280.030541
7      1476.624884
8      1460.769753
          ...     
490    1336.515307
493    1330.233179
497    1919.856304
498    1863.014814
499    1295.012183
Name: dollars, Length: 250, dtype: float64

In [56]:
New = cred_prom["dollars"].loc[cred_prom["insert"]=="New Promotion"]
New

1      1403.807542
4      1513.563200
5      1729.627996
6      1609.705918
9      1854.489028
          ...     
491    2227.464385
492    2007.541756
494    2000.050800
495    1513.060134
496    1020.757943
Name: dollars, Length: 250, dtype: float64

In [59]:
print("The average spent by the normal standard seasonal ad is ${}".format(standard.mean()))

The average spent by the normal standard seasonal ad is $1566.3890309659348


In [61]:
print("The average spent by the New Promotional ad is ${}".format(New.mean()))

The average spent by the New Promotional ad is $1637.4999830647992


In [64]:
print("The difference between mean of both spends based on ads recieved by customers is {}".format(New.mean()-standard.mean()))

The difference between mean of both spends based on ads recieved by customers is 71.11095209886435


- On average, the customers who recieved the interest-rate promotion charged about $ 71.11 more than the normal standard seasonal ad, and the vary a little more around their average. 

### Two Sample T-test (Independent)

### H0 : Sales Because of Standard Seasonal Ad  ==  Sales because of New Promotional Ad

### H1 : Sales Because of Standard Seasonal Ad  <  Sales because of New Promotional Ad

In [66]:
#Equal Variance i.e equal_var = True
eq_var = stats.ttest_ind(a= standard,
                         b=New,
                        equal_var = True)
eq_var.statistic

-2.2604227264649963

In [68]:
#Unequal Variance i.e equal_var = False
uneq_var = stats.ttest_ind(a= standard,
                         b=New,
                        equal_var = False)
uneq_var.statistic

-2.260422726464996

#### We will consider Equal_variance since T-score (statistic) is not having huge difference between both the tests

In [69]:
uneq_var.statistic - eq_var.statistic

4.440892098500626e-16

In [71]:
p = eq_var.pvalue
t = eq_var.statistic

print("For the above test, the t-score is {} and the p-value is {}".format(t,p))

if (p<0.05):
    print("We reject Null hypothesis")
else:
        print("We Fail to reject Null hypothesis")

For the above test, the t-score is -2.2604227264649963 and the p-value is 0.024225996894147814
We reject Null hypothesis


- Since the significant value of the test is less than 0.05, We can safely conclude that the average of 71.11 dollars more spent by cardholders recieving the reduced interest rate is not due to chance alone. The store will now consider extending the offers to all credit cutomers.

### 3. An experiment is conducted to study the hybrid seed production of bottle gourd under open field conditions. The main aim of the investigation is to compare natural pollination and hand pollination. The data are collected on 10 randomly selected plants from each of natural pollination and hand pollination. The data are collected on fruit weight (kg), seed yield/plant (g) and seedling length (cm). (Data set: pollination.csv)

In [92]:
Poll = pd.read_csv("E:Datasets/pollination.csv")
Poll

Group  Fruit_Wt  Seed_Yield_Plant  Seedling_length
0   Natural      1.85            147.70            16.86
1   Natural      1.86            136.86            16.77
2   Natural      1.83            149.97            16.35
3   Natural      1.89            172.33            18.26
4   Natural      1.80            144.46            17.90
5   Natural      1.88            138.30            16.95
6   Natural      1.89            150.58            18.15
7   Natural      1.79            140.99            18.86
8   Natural      1.85            140.57            18.39
9   Natural      1.84            138.33            18.58
10     Hand      2.58            224.26            18.18
11     Hand      2.74            197.50            18.07
12     Hand      2.58            230.34            19.07
13     Hand      2.62            217.05            19.00
14     Hand      2.68            233.84            18.00
15     Hand      2.56            216.52            18.49
16     Hand      2.34            211.93            17.45
17     Hand      2.67            210.37            18.97
18     Hand      2.45            199.87            19.31
19     Hand      2.44            214.30            19.36

### a.  Is the overall population of Seed yield/plant (g) equals to 200?


### One Sample T-test

- Method used : stats.ttest_1samp(a=, popmean= )
#### H0 : Population of Seed yield / plant == 200
#### H1 : Population of Seed yield / plant != 200


In [77]:
#Mean
Poll.Seed_Yield_Plant.mean()

180.8035

In [80]:
ttest_1 = stats.ttest_1samp(a= Poll.Seed_Yield_Plant, popmean= 200)
t_score = round(ttest_1.statistic, 2)
p_value = ttest_1.pvalue
print("The p value is {} and the t score is {}".format(p_value,t_score))

The p value is 0.032891040921283025 and the t score is -2.3


In [82]:
p_value > 0.05

False

- Since the significance value of the T-test is less than 0.05, We Reject Null hypothesis. Therefore overall population of Seed yield/plant is not equal to 200.

### b.  Test whether the natural pollination and hand pollination under open field conditions are equally effective or are significantly different.


### Two Sample T-test(Paired)
#### Method used: stats.ttest_rel(a=,b=)

### Stage-1

In [100]:
Natural_fruit_wt = Poll["Fruit_Wt"].loc[Poll["Group"] == "Natural"]
Hand_fruit_wt = Poll["Fruit_Wt"].loc[Poll["Group"] == "Hand"]

In [101]:
Natural_fruit_wt.mean()

1.848

In [104]:
print("The Mean of Natural Pollination of Fruit_wt is {}".format(Natural_fruit_wt.mean()))

The Mean of Natural Pollination of Fruit_wt is 1.848


In [102]:
Hand_fruit_wt.mean()

2.5660000000000003

In [105]:
print("The Mean of Hand Pollination of Fruit_wt is {}".format(Hand_fruit_wt.mean()))

The Mean of Hand Pollination of Fruit_wt is 2.5660000000000003


In [103]:
diff = Hand_fruit_wt.mean() - Natural_fruit_wt.mean()
diff

0.7180000000000002

#### H0 : Fruit_wt(Natural) = Fruit_wt(Hand)
#### H1 : Fruit_wt(Natural) <> Fruit_wt(Hand)

In [107]:
#Stage 1
ttest1 = stats.ttest_rel(a=Natural_fruit_wt,b=Hand_fruit_wt)


#Statistic : t score
t1 = ttest1.statistic

# P value
p1 = ttest1.pvalue

print("The t-score is {} and the p-value is {}".format(t1,p1))

The t-score is -16.003484109505287 and the p-value is 6.420421224715851e-08


In [108]:
p1 > 0.05

False

- Since p1 is smaller than 0.05 we reject Null hypothesis(H0)

### Stage-2

In [111]:
Natural_Seed_yield = Poll["Seed_Yield_Plant"].loc[Poll["Group"] == "Natural"]
Hand_Seed_yield = Poll["Seed_Yield_Plant"].loc[Poll["Group"] == "Hand"]

In [112]:
Natural_Seed_yield.mean()

146.009

In [117]:
print("The Mean Weight of Natural Pollination of Seed_yield_plant is {}".format(Natural_Seed_yield.mean()))

The Mean Weight of Natural Pollination of Seed_yield_plant is 146.009


In [118]:
Hand_Seed_yield.mean()

215.598

In [119]:
print("The Mean Weight of Hand Pollination of Seed_yield_plant is {}".format(Hand_Seed_yield.mean()))

The Mean Weight of Hand Pollination of Seed_yield_plant is 215.598


In [116]:
diff = Hand_Seed_yield.mean() - Natural_Seed_yield.mean()
diff

69.58900000000003

#### H0 : Seed_Yield_Plant(Natural) = Seed_Yield_Plant(Hand)
#### H1 : Seed_Yield_Plant(Natural) <>Seed_Yield_Plant(Hand)

In [120]:
#Stage 2
ttest2 = stats.ttest_rel(a=Natural_Seed_yield,b=Hand_Seed_yield)


#Statistic : t score
t2 = ttest2.statistic

# P value
p2 = ttest2.pvalue

print("The t-score is {} and the p-value is {}".format(t2,p2))

The t-score is -16.80079294985747 and the p-value is 4.198813240596638e-08


In [121]:
p2 > 0.05

False

- Since the p2 is lesser than 0.05 we reject Null hypotheis(H0)

### Stage-3

In [123]:
Natural_Seedling_length = Poll["Seedling_length"].loc[Poll["Group"] == "Natural"]
Hand_Seedling_length = Poll["Seedling_length"].loc[Poll["Group"] == "Hand"]

In [125]:
Natural_Seedling_length.mean()

17.706999999999994

In [127]:
print("The Mean of Natural_Seedling_length is {}".format(Natural_Seedling_length.mean()))

The Mean of Natural_Seedling_length is 17.706999999999994


In [126]:
Hand_Seedling_length.mean()

18.589999999999996

In [128]:
print("The Mean of Hand_Seedling_length is {}".format(Hand_Seedling_length.mean()))

The Mean of Hand_Seedling_length is 18.589999999999996


In [131]:
diff = Hand_Seedling_length.mean() - Natural_Seedling_length.mean()
diff

0.8830000000000027

#### H0 : Seedling_length(Natural) = Seedling_length(Hand)
#### H1 : Seedling_length(Natural) <>Seedling_length(Hand)

In [135]:
#Stage 3
ttest3 = stats.ttest_rel(a=Natural_Seedling_length,b=Hand_Seedling_length)


#Statistic : t score
t3 = ttest3.statistic

# P value
p3 = ttest3.pvalue

print("The t-score is {} and the p-value is {}".format(t3,p3))

The t-score is -2.9765905136151303 and the p-value is 0.015535315669362245


In [137]:
p3 > 0.05

False

- Since the p3 is lesser than 0.05 we reject Null hypothesis(H0)

### Conclusion

- Since we rejected Null Hypothesis in each stages we can confidently conclude that,the natural pollination and hand pollination under open field conditions are not equally effective but Significantly Hand poolination is more effective because,
- Average weight per Fruit is increased in Hand pollination, Average weight per Seed_Yield_Plant is increased in Hand Pollination, Average length per Seedling is increased in Hand Pollination

### 4. An electronics firm is developing a new DVD player in response to customer requests. Using a prototype, the marketing team has collected focus data for different age groups viz. Under 25; 25-34; 35-44; 45-54; 55-64; 65 and above. Do you think that consumers of various ages rated the design differently? (Data set: dvdplayer.csv).


In [139]:
dvd = pd.read_csv("E:Datasets/dvdplayer.csv")
dvd

agegroup   dvdscore
0   65 and over  38.454803
1         55-64  17.669677
2   65 and over  31.704307
3   65 and over  25.924460
4      Under 25  30.450007
..          ...        ...
63        45-54  46.567682
64  65 and over  23.999491
65     Under 25  24.994419
66  65 and over  33.538502
67        45-54  37.201829

[68 rows x 2 columns]

### F-test / Anova

- Method used : stats.f_oneway

In [141]:
age_group_1 = dvd['dvdscore'].loc[dvd['agegroup']=="65 and over"]
age_group_2 = dvd['dvdscore'].loc[dvd['agegroup']=="55-64"]
age_group_3 = dvd['dvdscore'].loc[dvd['agegroup']=="Under 25"]
age_group_4 = dvd['dvdscore'].loc[dvd['agegroup']=="35-44"]
age_group_5 = dvd['dvdscore'].loc[dvd['agegroup']=="45-54"]
age_group_6 = dvd['dvdscore'].loc[dvd['agegroup']=="25-34"]


#### Consumers of various age groups rated the design similarly :

#### H0 : Mean(age group_1(65 and over)) == mean(age group_2(55-64)) == mean(age group3(Under 25)) == mean(age group4(35-54))                 
#### == mean(age group5(45-54)) == mean(age group 6(25-34))
 
#### Consumers of various age groups rated the design differently :                     

#### Ha  : mean(age group 1(65 and over)) != mean(age group 2(55-64)) != mean(age group 3(Under 25))!= mean(age group 4(35-54)) != 

#### mean(age group 5(45-54)) != mean(age group 6(25-34))
                         
                         


In [144]:
# Perfrom the Anova
anova = stats.f_oneway(age_group_1,age_group_2,age_group_3,age_group_4,age_group_5,age_group_6)

# Statistic :  F Value
f = anova.statistic

#P-value
p = anova.pvalue
print("The f-value is {} and the p value is {}".format(f,p))
if(p<0.05):
    print('We reject null hypothesis')
else:
    print('We fail to reject null hypothesis')

The f-value is 6.992526962676517 and the p value is 3.087324905679639e-05
We reject null hypothesis


### Conclusion

* Since the significance value of the test is less than 0.05, we reject the null hypothesis. Therefore,consumers of various ages rated the design differently

### 5. A  survey  was  conducted  among  2800  customers  on  several  demographic characteristics. Working status, sex, age, age-group, race, happiness, no. of child, marital status, educational qualifications, income group etc. had been captured for that purpose. (Data set: sample_survey.csv).



In [146]:
sam_sur = pd.read_csv("E:Datasets/sample_survey.csv")
sam_sur

id            wrkstat        marital  childs   age  educ  paeduc  \
0        1  Working full time       Divorced     2.0  60.0  12.0    12.0   
1        2  Working part-time  Never married     0.0  27.0  17.0    20.0   
2        3  Working full time        Married     2.0  36.0  12.0    12.0   
3        4  Working full time  Never married     0.0  21.0  13.0     NaN   
4        5  Working full time  Never married     0.0  35.0  16.0     NaN   
...    ...                ...            ...     ...   ...   ...     ...   
2827  2828  Working full time        Married     2.0  73.0  14.0     8.0   
2828  2829  Working full time       Divorced     0.0  40.0  12.0    12.0   
2829  2830            Retired        Married     0.0  73.0   6.0     4.0   
2830  2831  Working full time  Never married     0.0  47.0  12.0    12.0   
2831  2832  Working part-time  Never married     0.0  20.0  10.0     NaN   

      maeduc  speduc          degree  ...        agecat childcat news1 news2  \
0       12.0     NaN     High school  ...      55 to 64      1-2    No    No   
1        NaN     NaN  Junior college  ...      25 to 34     None    No    No   
2       12.0    16.0     High school  ...      35 to 44      1-2    No    No   
3       12.0     NaN     High school  ...  Less than 25     None    No    No   
4       12.0     NaN        Bachelor  ...      35 to 44     None    No    No   
...      ...     ...             ...  ...           ...      ...   ...   ...   
2827    12.0    12.0  Junior college  ...   65 or older      1-2    No    No   
2828    12.0     NaN     High school  ...      35 to 44     None    No    No   
2829     NaN    10.0  LT High school  ...   65 or older     None   Yes    No   
2830    12.0     NaN     High school  ...      45 to 54     None    No    No   
2831     NaN     NaN  LT High school  ...  Less than 25     None    No    No   

      news3 news4 news5      car1      car2      car3  
0        No    No    No  American  Japanese  Japanese  
1       Yes    No    No  American    German  Japanese  
2        No   Yes   Yes  American  American       NaN  
3        No   Yes   Yes  American     Other       NaN  
4        No    No    No  American  American    Korean  
...     ...   ...   ...       ...       ...       ...  
2827     No    No    No  American  American    German  
2828     No    No    No  Japanese       NaN       NaN  
2829    Yes    No    No    Korean  American  American  
2830     No    No    No  Japanese       NaN       NaN  
2831     No    No    No   Swedish  American    German  

[2832 rows x 46 columns]

### a.   Is there any relationship in between labour force status with marital status?


### Chi-Square Test

#### H0 : Observed == Expected (No relation between labour force status and marital status)

#### Ha : Observed != Expected (There is a realtionship between labour force status and marital status)

In [151]:
#Creating Crosstab with workstatus and marital status of labours
lab_mar_xtab = pd.crosstab(sam_sur.wrkstat, sam_sur.marital, margins = True)
lab_mar_xtab

marital                  Divorced  Married  Never married  Separated  Widowed  \
wrkstat                                                                         
Keeping house                  25      200             35         13       55   
Other                          12       16             14          4        8   
Retired                        53      168             17          6      150   
School                          7        9             60          2        1   
Temporarily not working         9       23             11          1        2   
Unemployed, laid off           10       13             32          0        3   
Working full time             295      778            392         58       44   
Working part-time              35      138            102          9       20   
All                           446     1345            663         93      283   

marital                   All  
wrkstat                        
Keeping house             328  
Other                      54  
Retired                   394  
School                     79  
Temporarily not working    46  
Unemployed, laid off       58  
Working full time        1567  
Working part-time         304  
All                      2830

In [154]:
#Performing Chi2-test and getting all the respected values
x2test = stats.chi2_contingency(observed= lab_mar_xtab)

x2test

(729.2421426572284,
 1.820339965538765e-127,
 40,
 array([[5.16918728e+01, 1.55886926e+02, 7.68424028e+01, 1.07787986e+01,
         3.28000000e+01, 3.28000000e+02],
        [8.51024735e+00, 2.56643110e+01, 1.26508834e+01, 1.77455830e+00,
         5.40000000e+00, 5.40000000e+01],
        [6.20932862e+01, 1.87254417e+02, 9.23045936e+01, 1.29477032e+01,
         3.94000000e+01, 3.94000000e+02],
        [1.24501767e+01, 3.75459364e+01, 1.85077739e+01, 2.59611307e+00,
         7.90000000e+00, 7.90000000e+01],
        [7.24946996e+00, 2.18621908e+01, 1.07766784e+01, 1.51166078e+00,
         4.60000000e+00, 4.60000000e+01],
        [9.14063604e+00, 2.75653710e+01, 1.35879859e+01, 1.90600707e+00,
         5.80000000e+00, 5.80000000e+01],
        [2.46954770e+02, 7.44740283e+02, 3.67109894e+02, 5.14950530e+01,
         1.56700000e+02, 1.56700000e+03],
        [4.79095406e+01, 1.44480565e+02, 7.12197880e+01, 9.99010601e+00,
         3.04000000e+01, 3.04000000e+02],
        [4.46000000e+02, 1.345

#### x2test [0] = Chi-Square score, x2test [1] = p-value, x2test [2] = degrees of freedom, x2test [3] = Expected value

In [153]:
print("The chi square stat is {} and the p value is {}".format(x2test[0],x2test[1]))

The chi square stat is 729.2421426572284 and the p value is 1.820339965538765e-127


In [158]:
x2test[1] > 0.05

False

### Conclusion

- Since the significance value of the test is less than 0.05, we reject the null hypothesis [H0]. Therefore there is a relation between labour force status and marital status

### b. Do you think educational qualification is somehow controlling the marital status?

### Chi- Square Test

#### H0 : Observed == Expected (No relation between educational qualification and marital status)

#### Ha : Observed != Expected (There is a realtionship between educational qualification and marital status)

In [167]:
#Creating Crosstab with degree and marital status of labours
edu_mar_xtab = pd.crosstab(sam_sur.degree, sam_sur.marital, margins = True)
edu_mar_xtab

marital         Divorced  Married  Never married  Separated  Widowed   All
degree                                                                    
Bachelor              58      251            129         12       28   478
Graduate              29      123             41          3        9   205
High school          241      686            367         58      148  1500
Junior college        45      108             46          3        6   208
LT High school        70      174             77         17       92   430
All                  443     1342            660         93      283  2821

In [168]:
#Performing Chi2-test and getting all the respected values
x2test_b = stats.chi2_contingency(observed= edu_mar_xtab)

x2test_b

(122.68449020508541,
 7.424404099753273e-15,
 25,
 array([[  75.06345268,  227.39312301,  111.83268345,   15.75824176,
           47.95249911,  478.        ],
        [  32.19248493,   97.52215526,   47.9617157 ,    6.75824176,
           20.56540234,  205.        ],
        [ 235.55476781,  713.57674583,  350.9393832 ,   49.45054945,
          150.4785537 , 1500.        ],
        [  32.66359447,   98.94930876,   48.66359447,    6.85714286,
           20.86635945,  208.        ],
        [  67.52570011,  204.55866714,  100.60262318,   14.17582418,
           43.1371854 ,  430.        ],
        [ 443.        , 1342.        ,  660.        ,   93.        ,
          283.        , 2821.        ]]))

#### x2test_b [0] = Chi-Square score, x2test_b [1] = p-value, x2test_b [2] = degrees of freedom, x2test_b [3] = Expected value

In [157]:
print("The chi square stat is {} and the p value is {}".format(x2test_b[0],x2test_b[1]))

The chi square stat is 122.68449020508541 and the p value is 7.424404099753273e-15


In [160]:
x2test_b[1] > 0.05

False

### Conclusion

* Since the significance value of the test is less than 0.05, we reject the null hypothesis. Therefore there is educational qualification somehow controlling the marital status.

### c.   Is happiness is driven by earnings or marital status?

### Chi-Square Test

### Happiness w.r.t to earnings

#### H0 : Observed == Expected (No relation between happiness and earnings)

#### Ha : Observed != Expected (There is a realtionship between happiness and earnings)

In [170]:
#Creating Crosstab with happy and income of labours
hap_ear_xtab = pd.crosstab(sam_sur.happy, sam_sur.income, margins = True)
hap_ear_xtab

income         $1000 TO 2999  $10000 - 14999  $15000 - 19999  $20000 - 24999  \
happy                                                                          
Not too happy              7              39              33              40   
Pretty happy              20             107             119             155   
Very happy                 5              44              26              50   
All                       32             190             178             245   

income         $25000 or more  $3000 TO 3999  $4000 TO 4999  $5000 TO 5999  \
happy                                                                        
Not too happy             113              9              9              6   
Pretty happy              888             11             13             18   
Very happy                571              4             10             11   
All                      1572             24             32             35   

income         $6000 TO 6999  $7000 TO 7999  $8000 TO 9999  LT $1000   All  
happy                                                                       
Not too happy             14             12              9        11   302  
Pretty happy              13             21             30        13  1408  
Very happy                 6             14             19        11   771  
All                       33             47             58        35  2481

In [169]:
#Performing Chi2-test and getting all the respected values
x2test_ear = stats.chi2_contingency(observed= hap_ear_xtab)

x2test_ear

(178.95053061216427,
 7.234749067043371e-21,
 36,
 array([[   3.89520355,   23.12777106,   21.66706973,   29.82265216,
          191.35187424,    2.92140266,    3.89520355,    4.26037888,
            4.01692866,    5.72108021,    7.06005643,    4.26037888,
          302.        ],
        [  18.16041919,  107.82748892,  101.01733172,  139.04070939,
          892.1305925 ,   13.62031439,   18.16041919,   19.86295848,
           18.72793229,   26.67311568,   32.91575977,   19.86295848,
         1408.        ],
        [   9.94437727,   59.04474002,   55.31559855,   76.13663845,
          488.51753325,    7.45828295,    9.94437727,   10.87666264,
           10.25513906,   14.60580411,   18.0241838 ,   10.87666264,
          771.        ],
        [  32.        ,  190.        ,  178.        ,  245.        ,
         1572.        ,   24.        ,   32.        ,   35.        ,
           33.        ,   47.        ,   58.        ,   35.        ,
         2481.        ]]))

#### x2test_ear [0] = Chi-Square score, x2test_ear [1] = p-value, x2test_ear [2] = degrees of freedom, x2test_ear [3] = Expected value

In [164]:
print("The chi square stat of happiness w.r.t to earnings {} and the p value is {}".format(x2test_ear[0],x2test_ear[1]))

The chi square stat of happiness w.r.t to earnings 178.95053061216427 and the p value is 7.234749067043371e-21


### Happiness w.r.t to marital status

#### H0 : Observed == Expected
    (No relation between happiness and marital status)
    
#### Ha : Observed != Expected
    (There is a realtionship between happiness and marital status)

In [171]:
#Creating Crosstab with happy and marital of labours
hap_mar_xtab = pd.crosstab(sam_sur.happy, sam_sur.marital, margins = True)
hap_mar_xtab

marital        Divorced  Married  Never married  Separated  Widowed   All
happy                                                                    
Not too happy        72       71            108         30       59   340
Pretty happy        278      684            426         49      137  1574
Very happy           93      582            120         13       83   891
All                 443     1337            654         92      279  2805

In [172]:
#Performing Chi2-test and getting all the respected values
x2test_mar = stats.chi2_contingency(observed= hap_mar_xtab)

x2test_mar

(260.68943894182826,
 7.762777322980048e-47,
 15,
 array([[  53.6969697 ,  162.06060606,   79.27272727,   11.15151515,
           33.81818182,  340.        ],
        [ 248.58538324,  750.24527629,  366.98609626,   51.62495544,
          156.55828877, 1574.        ],
        [ 140.71764706,  424.69411765,  207.74117647,   29.22352941,
           88.62352941,  891.        ],
        [ 443.        , 1337.        ,  654.        ,   92.        ,
          279.        , 2805.        ]]))

#### x2test_mar [0] = Chi-Square score, x2test_mar [1] = p-value, x2test_mar [2] = degrees of freedom, x2test_mar [3] = Expected value

In [173]:
print("The chi square stat of happiness w.r.t to marital status {} and the p value is {}".format(x2test_mar[0],x2test_mar[1]))

The chi square stat of happiness w.r.t to marital status 260.68943894182826 and the p value is 7.762777322980048e-47


### Final conclusion

- As it can be clearly seen that chi_square score for (happiness w.r.t to marital status) i.e. 260.68 is more than chi_square score for (happiness w.r.t to marital earnings) i.e 178.95, therefore we can conclude that happiness is driven by marital status
